master/buildbot/db/buildsets.py

```diff
@@ -88,15 +92,41 @@ def thd(conn):
             ins = br_tbl.insert()
             for buildername in builderNames:
                 self.check_length(br_tbl.c.buildername, buildername)
-                res = conn.execute(ins,
-                    dict(buildsetid=bsid, buildername=buildername, priority=priority,
-                        claimed_at=0, claimed_by_name=None,
-                        claimed_by_incarnation=None, complete=0, results=-1,
-                        submitted_at=submitted_at, complete_at=None,
-                        triggeredbybrid=triggeredbybrid, startbrid=startbrid))
 
+                # If this builder is being merged, figure out what to merge into
+                mergeBrDict = brDictsToMerge.get(buildername, None)
+                if mergeBrDict:
+                    mergebrid = artifactbrid = brDictsToMerge[buildername]['brid']
```

Comment:
we should use for `artifactbrid` variable   `brDictsToMerge[buildername]['artifactbrid']` if it has a value if not `bird`  (this happens when the artifact is generated from another build)

Reply:
fixed